plugins/todo-backend/src/plugin.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createBackendPlugin,
+  coreServices,
+} from '@backstage/backend-plugin-api';
+import { CompoundEntityRef, parseEntityRef } from '@backstage/catalog-model';
+import { InputError } from '@backstage/errors';
+import express, { Router } from 'express';
+import {
+  getBearerToken,
+  parseFilterParam,
+  parseIntegerParam,
+  parseOrderByParam,
+} from './lib/utils';
+import { todoReaderServiceRef } from './service/TodoReaderService';
+import { TODO_FIELDS } from './service/types';
+
+/**
+ * The Todos plugin is responsible for aggregating todo comments within source.
+ * @alpha
+ */
+export const todosPlugin = createBackendPlugin({
+  pluginId: 'todo-backend',
+  register(env) {
+    env.registerInit({
+      deps: {
+        todoReader: todoReaderServiceRef,
+        http: coreServices.httpRouter,
+      },
+      async init({ http, todoReader }) {
```

Comment:
A good shortcut here, is to just do `http.add(createRouter(...))`, since the plugin should be a 1:1 of the old stuff anyway. Then refactors can be done separately. The backend framework's main focus is, after all, just to get the wiring together, not to affect how things are constructed once the wiring is complete.

Reply:
Smart idea, I'll do that. Thanks for the pointer!